core/src/main/java/org/testcontainers/containers/ParsedDockerComposeFile.java

```diff
@@ -29,7 +30,7 @@
     private final File composeFile;
 
     @Getter
-    private Set<String> dependencyImageNames = new HashSet<>();
+    private Map<String, Set<String>> dependencyImageNames = new HashMap<>();
```

Comment:
I wold suggest to name a map as per this convention:
https://stackoverflow.com/questions/2253453/how-should-i-name-a-java-util-map

Reply:
Good to know, have changed it to serviceNameToImageNames